R/system.R

```diff
@@ -1176,7 +1176,67 @@ getDBConnection <- function(type, host = NULL, port = "", databaseName = "", use
         }
         conn <- eval(parse(text=connstr))
       } else if (host != "") {
```

Comment:
Check subtype instead, which should take values "dsn", "conn_str_kv", or "conn_str_text".

Reply:
Updated as suggested.